### DISCUSSION TITLE
Backpressure for Redis sink

### DISCUSSION THREAD

**Discussion by darklajid** [vectordotdev/vector#21943] [Q&A] [answered]
I'm in the process of evaluating Vector as a replacement for a logstash instance I'm having. One of the things we do is dump events into a Redis list using their builtin support. Vector can do the same thing according to the documentation, but there's one piece missing to be a drop-in replacement for us.

Logstash allows us to configure this output ("sink") to be limited to a number of items in the Redis list, effectively allowing us to make sure that
* the Redis instance won't explode if the consumer dies
* give us backpressure in logstash

It seems Vector's Redis sink doesn't support this. Would there be any other way for me to make this work?

**Comment by jszwedko** [comment, marked as answer]
I see logstash supports this via a couple of parameters:

- https://www.elastic.co/guide/en/logstash/current/plugins-outputs-redis.html#plugins-outputs-redis-congestion_interval
- https://www.elastic.co/guide/en/logstash/current/plugins-outputs-redis.html#plugins-outputs-redis-congestion_threshold

I think this would need to be an enhancement to Vector's `redis` sink. Would you want to open a [feature request](https://github.com/vectordotdev/vector/issues/new?assignees=&labels=type%3A+feature&projects=&template=feature.yml)?

**Comment by sc68cal** [comment]
Hi,

I have come across this discussion and my usecase is similar. [I have opened a feature request](https://github.com/vectordotdev/vector/issues/23096).

I have a _small_ amount of Rust experience, so I don't know if I have the chops to implement the feature, but I will at least begin some research into what it would require.
